src/libraries/Native/Unix/Common/pal_io_common.h

```diff
@@ -0,0 +1,189 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#pragma once
+
+#include <stdlib.h>
+#include <assert.h>
+#include <poll.h>
+#include <pal_error_common.h>
+#include <pal_utilities.h>
+
+/**
+ * Our intermediate pollfd struct to normalize the data types
+ */
+typedef struct
+{
+    int32_t FileDescriptor;  // The file descriptor to poll
+    int16_t Events;          // The events to poll for
+    int16_t TriggeredEvents; // The events that triggered the poll
+} PollEvent;
+
+/**
+ * Constants passed to and from poll describing what to poll for and what
+ * kind of data was received from poll.
+ */
+typedef enum
+{
+    PAL_POLLIN = 0x0001,   /* non-urgent readable data available */
+    PAL_POLLPRI = 0x0002,  /* urgent readable data available */
+    PAL_POLLOUT = 0x0004,  /* data can be written without blocked */
+    PAL_POLLERR = 0x0008,  /* an error occurred */
+    PAL_POLLHUP = 0x0010,  /* the file descriptor hung up */
+    PAL_POLLNVAL = 0x0020, /* the requested events were invalid */
+} PollEvents;
+
+inline static int32_t Common_Read(intptr_t fd, void* buffer, int32_t bufferSize)
```

Comment:
Is the `static` really required? I do not think it should. Marking this `static` may actually create unnecessary code duplication.

Reply:
@jkotas I've followed pattern we used for other functions, probably not required though. We used it only once per library so I think this won't make a difference as is

Reply:
> We used it only once per library 

It is hard to tell by looking at the sources, and not enforced by anything. I can bet that there are places some of these functions are used more than once.

Reply:
> I've followed pattern we used for other functions

That does not make it right...

Reply:
I'll fix it